testing-framework/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2023 Polyphene.
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod error;
+
+use cid::{multihash::Code, Cid};
+use fil_builtin_actors_bundle::BUNDLE_CAR;
+use futures::executor::block_on;
+use fvm::{
+    engine::EnginePool,
+    executor::{ApplyKind, DefaultExecutor, Executor},
+    init_actor,
+    machine::{DefaultMachine, Manifest, NetworkConfig},
+    state_tree::{ActorState, StateTree},
+    system_actor,
+};
+use fvm_integration_tests::{
+    dummy::DummyExterns,
+    tester::{Account, IntegrationExecutor},
+};
+use fvm_ipld_blockstore::{Block, Blockstore, MemoryBlockstore};
+use fvm_ipld_car::load_car_unchecked;
+use fvm_ipld_encoding::{serde::Serialize, CborStore};
+use libsecp256k1::{PublicKey, SecretKey};
+use rand::SeedableRng;
+
+use error::{Error, WrapFVMError};
+use fvm_shared::{
+    address::Address, bigint::Zero, econ::TokenAmount, message::Message, state::StateTreeVersion,
+    version::NetworkVersion, ActorID, IPLD_RAW,
+};
+
+// TODO: document purpose.
+const EAM_ACTOR_ID: ActorID = 10;
+const NETWORK_VERSION: NetworkVersion = NetworkVersion::V18;
+const DEFAULT_BASE_FEE: u64 = 100;
+
+/// Main interface to test `Actor`s with Kythera.
+pub struct Tester {
+    // Builtin actors root Cid used in the Machine
+    builtin_actors: BuiltInActors,
+    // Custom code cid deployed by developer
+    code_cids: Vec<Cid>,
```

Comment:
What is the usage of this property ? It's only written right now. I can forsee maybe two of them:
1. External usage, let our user know what are the CID of its deployed actors, both test and target
2. Internal usage, ability to know what are the links between test and target actor

In both cases, I think that differentiating the target and test actors seems like a good idea.

Also, the second case I've written (which is not currently covered I think) seems interesting to have.

Reply:
glad you mention! I was going to ask it, I removed it meanwhile, tbh I'd suggest we remove it for now as we don't have a use yet (YAGNI) and add it when we need, wdyt?